src/components/Application.tsx

```diff
@@ -341,36 +362,80 @@ export class Application extends React.Component<Props, State> {
   // Internals
   //
 
-  private get detectionsForCurrentMode(): (beachfront.Job|beachfront.ProductLine)[] {
+  private updateMapMode() {
```

Comment:
Same question as above - just wondering how the mapMode fits in with the overall Meridian fixes.

Reply:
Since detections and frames were being passed as getters before, every refresh would cause them to be updated in child components. With my updates this was causing redraws anytime the map was moved. It's a good optimization as well, since detections and frames were being regenerated anytime the app state changed in any way. I think it's probably best practice to not pass getters into props that generate a new object with each call.